pkg/controller/idler/idler_controller.go

```diff
@@ -341,11 +336,13 @@ func findPodByName(idler *toolchainv1alpha1.Idler, name string) *toolchainv1alph
 	return nil
 }
 
-// nextPodToBeKilledAfter checks the start times of all the tracked pods in the Idler and returns the timeout left
+// nextPodToBeKilledAfter checks the start times of all the tracked pods in the Idler and returns `true` and the timeout left
 // for the next pod to be killed.
-func nextPodToBeKilledAfter(idler *toolchainv1alpha1.Idler) *time.Duration {
+// If there is no pod to kill, the func returns `false`, `0`
+func nextPodToBeKilledAfter(idler *toolchainv1alpha1.Idler) (bool, time.Duration) {
```

Comment:
IMO it's not up to this function to decide if there should be a requeue or not.
As a matter of fact there should be always a requeue. No matter what.

Reply:
I've reverted the changes here in https://github.com/codeready-toolchain/member-operator/commit/c36a02a9cf89b85a24394cd04e6a2a7140890c51